L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleHGCDigis.cc

```diff
@@ -106,12 +117,15 @@ void HGCalTriggerNtupleHGCDigis::initialize(TTree& tree,
   tree.Branch("hgcdigi_eta", &hgcdigi_eta_);
   tree.Branch("hgcdigi_phi", &hgcdigi_phi_);
   tree.Branch("hgcdigi_z", &hgcdigi_z_);
-  char branchname[50];
+  std::string bname;
+  auto withBX([&bname](char const* vname, int bx) -> char const* {
+    bname = std::string(vname) + "_BX" + to_string(bx);
+    return bname.c_str();
+  });
   for (unsigned int i = 0; i < digiBXselect_.size(); i++) {
-    sprintf(branchname, "hgcdigi_data_BX%u", digiBXselect_.at(i));
-    tree.Branch(branchname, &hgcdigi_data_.at(i));
-    sprintf(branchname, "hgcdigi_isadc_BX%u", digiBXselect_.at(i));
-    tree.Branch(branchname, &hgcdigi_isadc_.at(i));
+    int bxi = digiBXselect_[i];
```

Comment:
I used `int bxi` in my example because `digiBXselect_` was a vector of `int`. But now it should rather be `unsigned`